app/services/CheckoutService.scala

```diff
@@ -328,22 +330,53 @@ class CheckoutService(identityService: IdentityService,
         autoRenew = renewal.plan.charges.billingPeriod.isRecurring,
         fastForwardTermStartDate = currentVersionExpired // If the sub has expired, we need to shift the term dates forward
       )
+      val finalRenewCommand = applyPromoIfNecessary(contact, basicRenewCommand)
+      logger.info(s"Final renew command will be: ${finalRenewCommand}")
+      finalRenewCommand
+    }
 
-      val validPromotion = for {
+    def getValidPromotion(contact: Contact) = for {
         code <- renewal.promoCode.withLogging("promo code from client")
         deliveryCountryString <- contact.mailingCountry.withLogging("salesforce mailing country")
         deliveryCountry <- CountryGroup.countryByName(deliveryCountryString).withLogging("delivery country object")
         validPromotion <- promoService.validate[com.gu.memsub.promo.Renewal](code, deliveryCountry, renewal.plan.id).withLogging("validated promotion").toOption
-      } yield validPromotion
-      val withPromo = validPromotion.map(v => p.apply(v, prpId => catalog.paid.find(_.id == prpId).map(_.charges.billingPeriod).get, promoPlans)(renewCommand)).getOrElse(renewCommand)
-      zuoraService.renewSubscription(withPromo)
+    } yield validPromotion
+
+    def applyPromoIfNecessary(contact: Contact, basicRenewCommand: Renew): Renew = {
+      getValidPromotion(contact) match {
+        case Some(validPromotion) => {
+          logger.info(s"Attempting to apply the promotion: ${validPromotion.code} to subscription: ${subscription.name}")
+          p.apply(validPromotion, prpId => catalog.paid.find(_.id == prpId).map(_.charges.billingPeriod).get, promoPlans)(basicRenewCommand)
+        }
+        case None => {
+          logger.info(s"No promotion to apply, keeping basicRenewCommand")
```

Comment:
if the above implicit is defined, then using info(...) will insert the sub id into the log message for free